widgets/menu.js

```diff
@@ -0,0 +1,77 @@
+var React = require('react');
+var cx = require('classnames');
+
+var utils = require('../lib/widget-utils.js');
+
+var defaultTemplate = `<a href="{{href}}">{{name}}</a> {{count}}`;
+
+var hierarchicalCounter = 0;
+
+/**
+ * Instantiate a list of refinements based on a facet
+ * @param  {String|DOMElement} options.container Valid CSS Selector as a string or DOMElement
+ * @param  {String} options.facetName Name of the attribute for faceting
+ * @param  {String[]} [options.sortBy=['count:desc']] How to sort refinements. Possible values: `count|isRefined|name:asc|desc`
+ * @param  {String} [options.limit=100] How much facet values to get.
+ * @param  {String|String[]} [options.rootClass=null] CSS class(es) for the root `<ul>` element
+ * @param  {String|String[]} [options.itemClass=null] CSS class(es) for the item `<li>` element
+ * @param  {String|Function} [options.template] Item template, provided with `name`, `count`, `isRefined`
+ * @return {Object}
+ */
+function menu({
+    container = null,
+    facetName = null,
+    sortBy = ['count:desc'],
+    limit = 100,
+    rootClass = null,
+    itemClass = null,
+    template = defaultTemplate
+  }) {
+  hierarchicalCounter++;
+
+  var MultipleChoiceList = require('../components/MultipleChoiceList');
+
+  var containerNode = utils.getContainerNode(container);
+  var usage = 'Usage: menu({container, facetName, [sortBy, limit, rootClass, itemClass, template]})';
+
+  if (container === null || facetName === null) {
+    throw new Error(usage);
+  }
+
+  var hierarchicalFacetName = 'instantsearch.js' + hierarchicalCounter;
+
+  return {
+    getConfiguration: () => ({
```

Comment:
In fact here it's required by ES6, Sylvain asked the same question. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions

I agree this is a bit surprising.

"Because curly braces are used to denote the function’s body, an arrow function that wants to return an object literal outside of a function body must wrap the literal in parentheses. For example"

https://www.nczonline.net/blog/2013/09/10/understanding-ecmascript-6-arrow-functions/